qt/widgets/common/src/FunctionModel.cpp

```diff
@@ -487,6 +489,37 @@ void FunctionModel::updateGlobals() {
   }
 }
 
+void FunctionModel::setResolutionFromWorkspace(IFunction_sptr fun) {
+  auto n = fun->getNumberDomains();
+  if (n > 1) {
+    for (size_t index = 0; index < n; index++) {
+      setResolutionFromWorkspace(fun->getFunction(index));
+    }
+  } else {
+    if (fun->hasAttribute("f0.Workspace")) {
+      std::string wsName = fun->getAttribute("f0.Workspace").asString();
+      if (AnalysisDataService::Instance().doesExist(wsName)) {
+        const auto ws =
+            AnalysisDataService::Instance().retrieveWS<MatrixWorkspace>(wsName);
+        auto inst = ws->getInstrument();
+        auto analyser = inst->getStringParameter("analyser");
+        if (analyser.size() > 0) {
```

Comment:
With the new extracted method we could do an early return with
```cpp
if(analyser.empty())
  return
```
to further reduce the indentation and improve readability?